File: library/src/main/java/net/placemarkt/RegexPatternCache.java
package net.placemarkt;

import java.util.Map;
import java.util.HashMap;
import java.util.regex.Pattern;

class RegexPatternCache {
  private final Map<String, Pattern> map = new HashMap<>();

  public Pattern get(String key) {
    return map.computeIfAbsent(key, Pattern::compile);
  }
}

File: library/src/main/java/net/placemarkt/RegexPatternCache.kt
package net.placemarkt

internal class RegexPatternCache {
    private val map: MutableMap<String, Regex> = hashMapOf()

    operator fun get(
        key: String,
        vararg regexOptions: RegexOption,
        ): Regex = map.computeIfAbsent(key) { Regex(it, regexOptions.toSet()) }
}
